examples/custom-vnet/main.tf

```diff
@@ -57,4 +57,13 @@ module "lacework_azure_agentless_scanning_rg_and_vnet" {
   custom_network                 = "" // tolist(azurerm_virtual_network.example.subnet)[0].id
   create_log_analytics_workspace = true
   region                         = local.region
+
+  // Creating a NAT gateway with a custom vnet requires the network security group
+  // to be specified here in the custom_network_security_group variable. If you do
+  // not wish to use a NAT gateway, do not use this variable and instead set use_nat_gateway
+  // to false. 
+
+  // Only uncomment one or the other!
+  // custom_network_security_group = azurerm_network_security_group.example.id
+  // use_nat_gateway = false
```

Comment:
nit: Perhaps we should set `custom_network_security_group` by default, aligning with the idea of "convention over configuration".

I understand that the intention of leaving both `use_nat_gateway` and `custom_network_security_group` commented here is to clarify that they're mutually exclusive. 

However, since the default is `use_nat_gateway = true`, does it make sense to have `custom_network_security_group` set by default to align with that? In this case, if someone wants to disable the NAT gateway, they will also unset `custom_network_security_group` — otherwise, the validation will trigger and clarify that only one of these two variables should be set at a time. 

We could tweak the comments to remain explicit while still defaulting to the most common case.
```suggestion
  // When using a custom vnet with the default NAT gateway (use_nat_gateway = true),
  // you must specify the network security group here:
  custom_network_security_group = azurerm_network_security_group.example.id

  // If you want to use public IPs instead of a NAT gateway, comment out the line above
  // and uncomment this line:
  // use_nat_gateway = false
```

IMO, this approach provides better developer experience for the most common use case as it would work out of the box, and we would still retain safety provided by the validation rules you added.

What do you think?

Reply:
I like it! 

The only reason I had both options commented out by default was because above on line 57 we also leave the `custom_network` commented out. I presumed this was because trying to run `terraform apply` will result in an error if the vnet doesn't already exist. So I was just trying to follow the precendent.

But on further consideration I don't really love that -- it's confusing and the customer has to run `terraform apply` twice regardless. 

So TL;DR I've committed your suggestion, but also uncommented `custom_network` above.